libraries/AP_HAL_Linux/RCInput_RPI.h

```diff
@@ -131,12 +131,15 @@ class RCInput_RPI : public RCInput
     } rc_channels[RCIN_RPI_CHN_NUM];
 
     bool _initialized = false;
-
+    int _version =0;
```

Comment:
I may be missing something but this seems to always be 0 since it's not set anywhere. We should rather get the value returned from the utils class that already knows the SoC version.

Reply:
No it is my mistake... I initialized it:
`_version = UtilRPI::from(hal.util)->get_rpi_version();`

Now another question... should we refactor 
**set_physical_addresses** and version from parameter? as I had to use _version as a class member because **init_PCM** is called twice in code in two places and I think in **_timer_tick()** we need every thing as fast as possible.

Reply:
The second call to init_PCM() from timer_tick only happens when there is an error. See the debug message there. Are you seeing that message when running?

Reply:
No I dont see errors.... what I am saying here is that I created a class member _version instead of calling 
`UtilRPI::from(hal.util)->get_rpi_version();  `
in multiple places. and because I did so I am asking if we need to remove version from 
`void RCInput_RPI::set_physical_addresses(int version)`
and make it
`void RCInput_RPI::set_physical_addresses()`
and use _version directly inside the function.....
anyway It is another enhancement. If you are ok with it you can proceed.